main/src/ca/uwaterloo/flix/language/phase/optimizer/Inliner.scala

```diff
@@ -371,6 +373,195 @@ object Inliner {
       Expr.NewObject(name, clazz, tpe, eff, methods, loc)
   }
 
+  /**
+    * Evaluate match if possible.
```

Comment:
More precise? Like how much evaluation does it do? For example, does it simplify a match?

e.g.

```
match Some(UNKNOWN) { ... } 
```

I suppose not, having not read further, so I would say something:

Statically evaluates a match expression to one of its match rules if the scrutinee is known (or sufficiently known?)

Reply:
I've added the overall logic of the function. I think documenting all details will just go out of date immediately and not give much benefit.